src/devices/machine/i3001.h

```diff
@@ -64,11 +64,11 @@ class i3001_device : public device_t
 	DECLARE_READ_LINE_MEMBER(zero_r) { return m_zero; }
 
 	// Output callbacks
-	auto fo_w() { return m_fo_handler.bind(); }
+	template <typename... T> void set_fo_w_cb(T &&... args) { m_fo_handler.set(std::forward<T>(args)...); }
 
 	// Input callbacks
-	auto px_r() { return m_px_handler.bind(); }
-	auto sx_r() { return m_sx_handler.bind(); }
+	template <typename... T> void set_px_r_cb(T &&... args) { m_px_handler.set(std::forward<T>(args)...); }
+	template <typename... T> void set_sx_r_cb(T &&... args) { m_sx_handler.set(std::forward<T>(args)...); }
```

Comment:
These are configuration methods between live methods.  Please put all the configuration methods at the top of the class declaration.